lnd.go

```diff
@@ -667,25 +667,51 @@ func Main(cfg *Config, lisCfg ListenerCfg, implCfg *ImplementationCfg,
 	ltndLog.Infof("Waiting for chain backend to finish sync, "+
 		"start_height=%v", bestHeight)
 
-	for {
-		if !interceptor.Alive() {
+	type syncResult struct {
+		IsSynced bool
+		TS       int64
+		Err      error
+	}
+
+	syncedResChan := make(chan syncResult, 1)
+	isSynced := false
+
+	for !isSynced {
```

Comment:
I think we can get away without the `isSynced` variable. Something like:

```
        var syncedResChan = make(chan syncResult, 1)
	for {
		// We check if the wallet is synced in a separate goroutine as
		// the call is blocking, and we want to be able to interrupt it
		// if the daemon is shutting down.
		go func() {
			synced, bestBlock, err := activeChainControl.Wallet.
				IsSynced()
			syncedResChan <- syncResult{synced, bestBlock, err}
		}()

		select {
		case <-interceptor.ShutdownChannel():
			return nil

		case res := <-syncedResChan:
			if res.err != nil {
				return mkErr("unable to determine if wallet "+
					"is synced: %v", res.err)
			}

			if res.synced {
				break
			}

			// If we're not yet synced, we'll wait for a second
			// before checking again.
			select {
			case <-interceptor.ShutdownChannel():
				return nil

			case <-time.After(time.Second):
				continue
			}
		}

		break
         }
	
```

Reply:
Thanks! Updated the PR to your suggestion :rocket:. I'm in sight favour of the `isSynced` variable approach though as I think that's a bit more readable, but have no strong opinion :). I did however keep the "Syncing to block timestamp" log line, as I do think that adds value for users.

Reply:
>  I did however keep the "Syncing to block timestamp" log line, as I do think that adds value for users.

yup - was just posting pseudo code to show the design, not an exact "pls copy as is" snippet